Assets/Scripts/CommandPattern/InputHandler.cs

```diff
@@ -1,49 +0,0 @@
-﻿using UnityEngine;
-
-public class InputHandler : GameSceneController, ICameraComponent
-{
-    BattleSystemController battleSystem;
```

Comment:
Your dependencies should be interfaces.
You can even expose them in the inspector with a trick (this is one of the solutions out there https://www.patrykgalach.com/2020/01/27/assigning-interface-in-unity-inspector/)